dap_aria_mapping/notebooks/novelty/pipeline_openalex_novelty.py

```diff
@@ -0,0 +1,81 @@
+""" 
+Script/pipeline to calculate novelty scores for OpenAlex papers
+Uses typer to create a command line interface
+
+Usage examples:
+
+All levels, full dataset:
+python dap_aria_mapping/notebooks/novelty/pipeline_openalex_novelty.py
+
+One level, test dataset:
+python dap_aria_mapping/notebooks/novelty/pipeline_openalex_novelty.py --taxonomy-level 1 --test
+"""
+from dap_aria_mapping import logging, PROJECT_DIR
+import dap_aria_mapping.getters.openalex as oa
+import dap_aria_mapping.utils.novelty_utils as nu
+import typer
+
+OUTPUT_FOLDER = PROJECT_DIR / "outputs/interim"
+
+
+def main(
+    taxonomy_level: int = 0,
+    test: bool = False,
+    n_test_sample: int = 1000,
+):
+    """
+    Calculate novelty scores for OpenAlex papers
+
+    Args:
+        taxonomy_level (int): Taxonomy level to use for novelty calculation. 
+            Must be between 1 and 5. If set to 0, uses all levels
+        test (bool): Whether to run in test mode (using a small sample of papers)
+        n_test_sample (int): Number of papers to use for testing
+
+    Returns:
+        None (saved novelty scores to file)
+    """
+    # Fetch OpenAlex metadata
```

Comment:
Is there a reason we're only doing this for papers? can we do it for patents as well?

Reply:
I haven't gotten to patent data yet - I would like to keep it for another PR. I think I need to first sense-check the results for papers, see if the novelty score actually yields any interesting results and insights.